From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 21 Jul 2022 17:38:00 +0200
Subject: [PATCH] Set SONAME in shared library

With `shared=yes` mupdf builds a shared library for libmupdf. Sadly,
this shared library does not have an SONAME set, since we setup up
versioned symlinks for the library we want it to have an SONAME. Should
look into upstreaming this eventually.

diff -upr mupdf-1.23.3.orig/Makefile mupdf-1.23.3/Makefile
--- mupdf-1.23.3.orig/Makefile	2023-09-09 10:37:11.676855234 +0200
+++ mupdf-1.23.3/Makefile	2023-09-09 10:43:39.114732726 +0200
@@ -85,7 +85,7 @@ $(OUT)/%.exe: %.c
 	$(LINK_CMD)
 
 $(OUT)/%.$(SO):
-	$(LINK_CMD) $(LIB_LDFLAGS) $(THIRD_LIBS) $(LIBCRYPTO_LIBS)
+	$(LINK_CMD) $(LIB_LDFLAGS) -Wl,-soname,$(notdir $@).$(SOVERSION) $(THIRD_LIBS) $(LIBCRYPTO_LIBS)
 
 $(OUT)/%.def: $(OUT)/%.$(SO)
 	$(GENDEF_CMD)
@@ -409,7 +409,11 @@ $(OUT)/storytest: docs/examples/storytes
 
 # --- Update version string header ---
 
-VERSION = $(shell git describe --tags)
+VERSION = $(shell test -d .git \
+	&& git describe --tags 2>/dev/null \
+	|| sed -n 's/.* FZ_VERSION "\([^"]\+\)"/\1/p' include/mupdf/fitz/version.h)
+
+SOVERSION = $(basename $(VERSION))
 
 version:
 	sed -i~ -e '/FZ_VERSION /s/".*"/"'$(VERSION)'"/' include/mupdf/fitz/version.h
@@ -448,7 +452,7 @@ install-libs: libs
 	install -m 644 include/mupdf/pdf/*.h $(DESTDIR)$(incdir)/mupdf/pdf
 ifneq ($(LIBS_TO_INSTALL_IN_LIB),)
 	install -d $(DESTDIR)$(libdir)
-	install -m 644 $(LIBS_TO_INSTALL_IN_LIB) $(DESTDIR)$(libdir)
+	install -m 755 $(LIBS_TO_INSTALL_IN_LIB) $(DESTDIR)$(libdir)
 endif
 
 install-apps: apps
